hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/TransitRegionStateProcedure.java

```diff
@@ -372,14 +372,12 @@ private Flow confirmOpened(MasterProcedureEnv env, RegionStateNode regionNode)
   }
 
   private void closeRegionAfterUpdatingMeta(MasterProcedureEnv env, RegionStateNode regionNode) {
-    CloseRegionProcedure closeProc =
+    LOG.debug("Close region: isSplit: {}: evictOnClose: {}", isSplit,
       isSplit
-        ? new CloseRegionProcedure(this, getRegion(), regionNode.getRegionLocation(),
-          assignCandidate,
-          env.getMasterConfiguration().getBoolean(EVICT_BLOCKS_ON_SPLIT_KEY,
-            DEFAULT_EVICT_ON_SPLIT))
-        : new CloseRegionProcedure(this, getRegion(), regionNode.getRegionLocation(),
-          assignCandidate, evictCache);
+        ? env.getMasterConfiguration().getBoolean(EVICT_BLOCKS_ON_SPLIT_KEY, DEFAULT_EVICT_ON_SPLIT)
+        : evictCache);
+    CloseRegionProcedure closeProc = new CloseRegionProcedure(this, getRegion(),
```

Comment:
Let's put a comment explaining this change:
`Splits/Merges are special cases, rather then deciding on the cache eviction behaviour here at Master, we just need to tell this close is for a split/merge and let RSes decide on the eviction. See HBASE-28811 for more context.`